UI/src/controller/massTransitStopsCollection.js

```diff
@@ -117,6 +139,17 @@
       getValidityPeriods: function() {
         return validityPeriods;
       },
+      getShowHideServicePoints: function() {
+        return showServicePoints;
+      },
+      showHideServicePoints: function(checked) {
+        if(!showServicePoints && checked){
+          showServicePoints = checked;
+          eventbus.trigger('servicePointCheckbox:changed', showServicePoints);
+        }
+        showServicePoints = checked;
+        eventbus.trigger('validityPeriod:changed', selectedValidityPeriods(validityPeriods));
+      },
```

Comment:
Sugestion to avoid showServicePoint = checked twice

```js 
    showHideServicePoints: function(checked) {
      showServicePoints = checked;
  
      if(checked){          
          eventbus.trigger('servicePointCheckbox:changed', showServicePoints);
        }
 
        eventbus.trigger('validityPeriod:changed', selectedValidityPeriods(validityPeriods));
      } ```

Reply:
Suggestion applied but with only one modification:
```javascript
      showHideServicePoints: function(checked) {
        if (checked) {
          eventbus.trigger('servicePointCheckbox:changed', showServicePoints);
        }

        showServicePoints = checked;
        eventbus.trigger('validityPeriod:changed', selectedValidityPeriods(validityPeriods));
      }